doc-generator.py

```diff
@@ -28,12 +28,25 @@ def format_doc(i, x):
             end_glyph = "--!"
             if not (start_glyph in x and end_glyph in x):
                 break
-            start = x.index(start_glyph)
-            end = x.index(end_glyph)
-            block = x[start+len(start_glyph):end]
-            formatted = highlight(block).strip()
-            x = x[:start] + formatted + x[end+len(end_glyph):]
-            print(x)
+            block_start = x.index(start_glyph)
+            cmd = x[block_start+len(start_glyph):x.index("\n", block_start)]
+            block_end = x.index(end_glyph)
+            block = x[block_start+len(start_glyph)+len(cmd):block_end]
+            if cmd == "code":
+                formatted = highlight(block).strip()
+                x = x[:block_start] + formatted + x[block_end+len(end_glyph):]
+            elif cmd == "params":
+                params = []
+                param_start = block.index("\n[")
+                while param_start != -1:
+                    param_end = block.index("]", param_start)
+                    param = block[param_start+len("\n["):param_end]
+                    param_start = block.find("\n[", param_end)
```

Comment:
This language is simple enough to be recognized by a regex. Maybe:
`r"\[(.*)\]\W*(.*)"`? (I didn't test it so make sure it works)